[c/en] mostly word clarification and extra descriptions (#4222)
adambard/learnxinyminutes-docs: c.html.markdown

// Enumeration constants are also ways to declare constants.
// All statements must end with a semicolon
enum days {SUN, MON, TUE, WED, THU, FRI, SAT};
// SUN gets 0, MON gets 1, TUE gets 2, etc.

// Enumeration values can also be specified
enum days {SUN = 1, MON, TUE, WED = 99, THU, FRI, SAT};
// MON gets 2 automatically, TUE gets 3, etc.
// WED get 99, THU gets 100, FRI gets 101, etc.

// Import headers with #include
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

// File names between <angle brackets> tell the compiler to look in your system
// libraries for the headers.
// For your own headers, use double quotes instead of angle brackets, and
// provide the path:
#include "my_header.h" 		// local file
#include "../my_lib/my_lib_header.h" //relative path

// Declare function signatures in advance in a .h file, or at the top of
// your .c file.
void function_1();
int function_2(void);

// At a minimum, you must declare a 'function prototype' before its use in any function.

> // Normally, protoypes are placed at the top of a file before any function definition.

Reviewer: ```suggestion
// Normally, prototypes are placed at the top of a file before any function definition.
```

int add_two_ints(int x1, int x2); // function prototype
// although `int add_two_ints(int, int);` is also valid (no need to name the args),
// it is recommended to name arguments in the prototype as well for easier inspection

> // Function protoypes are not necessary if the function definition comes before

Reviewer: ```suggestion
// Function prototypes are not necessary if the function definition comes before
```

// any other function that calls that function. However, it's standard practice to
// always add the function prototype to a header file (*.h) and then #include that
// file at the top. This prevents any issues where a function might be called

> // before the compiler knows of its existence; while also giving the developer a

Reviewer: ```suggestion
// before the compiler knows of its existence, while also giving the developer a
```

// clean header file to share with the rest of the project.

// Your program's entry point is a function called "main". The return type can

> // be anything, however most operating systems expect a return type of 'int' for

Reviewer: ```suggestion
// be anything, however most operating systems expect a return type of `int` for
```

  // For the sake of the tutorial, variables are declared dynamically under
  // C99-compliant standards.

>   // ints are usually 4 bytes (use the 'sizeof' operator to check)

Reviewer: ```suggestion
  // ints are usually 4 bytes (use the `sizeof` operator to check)
```

Author: i didn't realize that markdown would format code fences... inside of a code fence.

Reviewer: I actually don't know it for sure, but it's easier to discern the code that way, on my opinion :)

  int x_int = 0;

>   // shorts are usually 2 bytes (use the 'sizeof' operator to check)

Reviewer: ```suggestion
  // shorts are usually 2 bytes (use the `sizeof` operator to check)
```

  short x_short = 0;

  // chars are defined as the smallest addressable unit for a processor. 

>   // This is usually 1 byte, but some systems are more (ex. TMS320 from TI is 2 bytes).

Reviewer: ```suggestion
  // This is usually 1 byte, but for some systems it can be more (ex. for TMS320 from TI it's 2 bytes).
```
